client/src/main/java/com/boku/auth/http/client/BokuAPIClient.java

```diff
@@ -342,14 +353,35 @@ private AuthorizationHeader getSignedAuthorizationHeader() {
 
     }
 
-    private void verifyResponseSignature(HttpResponse httpResponse, BokuAPIClientResponse apiClientResponse) throws BokuAPIClientException {
+    private void verifyResponseSignature(HttpResponse httpResponse, BokuAPIClientResponse apiClientResponse, boolean requireSignedResponse)
+            throws BokuAPIClientException {
         Header[] respAuthHeaders = httpResponse.getHeaders(AuthorizationHeader.RESPONSE_HEADER);
-        if (respAuthHeaders.length != 1) {
-            throw new BokuAPIClientException(
-                "Got " + httpResponse.getStatusLine() + " with " + respAuthHeaders.length + " " + AuthorizationHeader.RESPONSE_HEADER + " headers, expected 1!",
-                apiClientResponse
-            );
+        if(requireSignedResponse) {
+            if (respAuthHeaders.length != 1) {
+                throw new BokuAPIClientException(
+                        "Got " + httpResponse.getStatusLine() + " with " + respAuthHeaders.length + " " + AuthorizationHeader.RESPONSE_HEADER + " headers, expected 1!",
+                        apiClientResponse
+                );
+            }
+
+            verifySignature(httpResponse, apiClientResponse, respAuthHeaders);
+        } else {
+            if(respAuthHeaders.length == 0){
```

Comment:
Another suggestion, as I don't normally like early returns like that.
```suggestion
            if(respAuthHeaders.length == 1){
                verifySignature(httpResponse, apiClientResponse, respAuthHeaders);
            } else if (respAuthHeaders.length > 1) {
                throw
            }
```